spring-tx/src/main/java/org/springframework/dao/support/DataAccessUtils.java

```diff
@@ -56,6 +60,89 @@
 		return results.iterator().next();
 	}
 
+	/**
+	 * Return a single result object from the given Stream.
+	 * <p>Returns {@code null} if 0 result objects found;
+	 * throws an exception if more than 1 element found.
+	 * @param results the result Stream (can be {@code null})
+	 * @return the single result object, or {@code null} if none
+	 * @throws IncorrectResultSizeDataAccessException if more than one
+	 * element has been found in the given Stream
+	 */
+	@Nullable
+	public static <T> T singleResult(@Nullable Stream<T> results) throws IncorrectResultSizeDataAccessException {
+		if (results == null) {
+			return null;
+		}
+		try (results) {
+			List<T> resultList = results.limit(2).toList();
+			if (resultList.size() > 1) {
+				throw new IncorrectResultSizeDataAccessException(1, resultList.size());
+			}
+			return CollectionUtils.isEmpty(resultList) ? null : resultList.get(0);
+		}
+	}
+
+	/**
+	 * Return a single result object from the given Iterator.
+	 * <p>Returns {@code null} if 0 result objects found;
+	 * throws an exception if more than 1 element found.
+	 * @param results the result Iterator (can be {@code null})
+	 * @return the single result object, or {@code null} if none
+	 * @throws IncorrectResultSizeDataAccessException if more than one
+	 * element has been found in the given Iterator
+	 */
+	@Nullable
+	public static <T> T singleResult(@Nullable Iterator<T> results) throws IncorrectResultSizeDataAccessException {
+		if (results == null) {
+			return null;
+		}
+		T result = results.hasNext() ? results.next() : null;
+		if (results.hasNext()) {
+			throw new IncorrectResultSizeDataAccessException(1, 2);
```

Comment:
Same suggestion as above : [Link](https://github.com/spring-projects/spring-framework/pull/27735/files#r757849057)
```suggestion
			throw new IncorrectResultSizeDataAccessException(1);
```
Since `actualSize` of the iterator isn't known beforehand, we can use `public IncorrectResultSizeDataAccessException(int expectedSize)` constructor and set the `actualSize`  as `-1`